esphome/components/haier/haier.cpp

```diff
@@ -0,0 +1,286 @@
+#include <cmath>
+#include "haier.h"
+#include "esphome/core/macros.h"
+
+namespace esphome {
+namespace haier {
+
+static const char *const TAG = "haier";
+
+void HaierClimate::dump_config() {
+  ESP_LOGCONFIG(TAG, "Haier:");
+  ESP_LOGCONFIG(TAG, "  Update interval: %u", this->get_update_interval());
+  dump_traits_(TAG);
```

Comment:
```suggestion
  this->dump_traits_(TAG);
```
In general there seems to be a `this->` prefix in some places but not in others, it's nice to be consistent (general ESPHome style is to use `this->`).